File: Staircase.js
function staircase(n) {
  let str = ""
  for (let i = 1; i <= n; i++) {
    console.log(str.padStart((n - i), " ") + str.padEnd(i, "#"))
  }
}

staircase(6)
staircase(20)

staircase(200)

File: Staircase.clj
(defn step [d n]
  (clojure.string/join (concat
                        (repeat (- n d) " ")
                        (repeat d "#"))))

(defn steps [n]
  (for [d (range 1 (inc n))]
    (step d n)))

(defn staircase [n]
  (steps n))

(println (staircase 6))
